set.mm

```diff
@@ -16950,24 +16950,39 @@ variables and no wff metavariables (see ~ ax12wdemo for an example of
   $}
 
   ${
-    $d x y $.  $d t u $.  $d z v w $.
-    $( Generalization of ~ hbaev , proved at no extra cost.  Instance of
-       ~ aev2 .  (Contributed by Wolf Lammen, 22-Mar-2021.)  (Revised by BJ,
-       29-Mar-2021.) $)
-    hbaevg $p |- ( A. x x = y -> A. z A. t t = u ) $=
-      ( vv vw weq wal aevlem alrimiv syl ) ABHAIFGHFIZEDHEIZCIABFGJMNCFGEDJKL
+    $v s $.  $d w s z $.
+    $( Define temporary individual variable s. $)
+    v.vs $f setvar s $.
+    $d x y $.  $d w s $.
+    $( A version of ~ aev with two universal quantifiers in the consequent.
+       One can prove similar statements with arbitrary numbers of universal
+       quantifiers in the consequent (the series begins with ~ aeveq , ~ aev ,
+       ~ aev2 ).
+
+       Using ~ aev and ~ alrimiv , one can actually prove (with no more axioms)
+       any scheme of the form ` ( A. x x = y -> ` PHI) , DV ` ( x , y ) ` where
+       PHI involves only setvar variables and the connectors ` -> ` , ` <-> ` ,
+       ` /\ ` , ` \/ ` , ` T. ` , ` = ` , ` A. ` , ` E. ` , ` E* ` , ` E! ` ,
+       ` F/ ` .  An example is given by ~ aevdemo .  This list cannot be
+       extended to ` -. ` or ` F. ` since the scheme ` A. x x = y ` is
+       consistent with ~ ax-mp , ~ ax-gen , ~ ax-1 -- ~ ax-13 (as the
+       one-element universe shows).
+
+       (Contributed by BJ, 23-Mar-2021.) $)
+    aev2 $p |- ( A. x x = y -> A. z A. t u = v ) $=
+      ( vw v.vs weq wal aev alrimiv syl ) ABIAJGHIGJZEDIFJZCJABGHGKNOCGHFDEKLM
       $.
-    $( $j usage 'hbaevg' avoids 'ax-8' 'ax-9' 'ax-10' 'ax-11' 'ax-12'
+    $( $j usage 'aev2' avoids 'ax-8' 'ax-9' 'ax-10' 'ax-11' 'ax-12'
        'ax-13'; $)
   $}
 
   ${
     $d x y $.
     $( Version of ~ hbae with a disjoint variable condition, requiring fewer
-       axioms.  Instance of ~ hbaevg and ~ aev2 .  (Contributed by Wolf Lammen,
+       axioms.  Instance of ~ aev2 .  (Contributed by Wolf Lammen,
        22-Mar-2021.) $)
     hbaev $p |- ( A. x x = y -> A. z A. x x = y ) $=
-      ( hbaevg ) ABCBAD $.
+      ( aev2 ) ABCBAAD $.
     $( $j usage 'hbaev' avoids 'ax-8' 'ax-9' 'ax-10' 'ax-11' 'ax-12'
        'ax-13'; $)
   $}
```

Comment:
What is the need to keep `hbaev`?

Reply:
(same question for `hbnaev`)

Reply:
Ok, I did not see that those have the `v.vs` hypothesis in the scope, while the `hb` theorems, not.